### ISSUE TITLE
Support YARD's ordered list syntax: Array<(String, Fixnum, Hash)>

### ISSUE THREAD

**Issue by connorshea** [AaronC81/sord#37]
**Is your feature request related to a problem? Please describe.**
YARD supports [a syntax involving parenthesis for Arrays that have fixed, ordered values](https://www.rubydoc.info/gems/yard/file/docs/Tags.md#Order-Dependent_Lists): `Array<(String, Fixnum, Hash)>`

Input:

```ruby
# @return [Array(Integer, Integer)] the current shard key
attr_reader :shard_key
```

Current Sord output:

```ruby
# sord warn - "Array(Integer, Integer)" does not appear to be a type
sig { returns(SORD_ERROR_ArrayIntegerInteger) }
def shard_key(); end
```

**Describe the solution you'd like**
Sord should be able to understand this syntax and turn it into a Sorbet type.

I think Tuples fit here as the Sorbet equivalent: https://sorbet.org/docs/tuples

>This creates a fixed array type (also referred to as a tuple), which is a fixed-length array with known types for each element. For example, `[String, T.nilable(Float)]` validates that an object is an array of exactly length 2, with the first item being a `String` and the second item being a `Float` or `nil`.

So given this input:
```ruby
# @return [Array(Integer, Integer)] the current shard key
attr_reader :shard_key
```

You'd want this output:

```ruby
sig { returns([Integer, Integer]) }
def shard_key(); end
```

The YARD documentation mentions only `Array<(String, Integer)>`, but it seems like `Array(String, Integer)` is also perfectly valid. The YARD types parser seems to handle this weirdly (but also that repo hasn't had a commit in nearly 10 years, so it may not be the best thing to work off of here).

**Comment by connorshea** [comment]
While implementing this, I noticed that a lot of types are now being assumed to be strings by YAML.load.

```ruby
# Check for literals
from_yaml = YAML.load(yard) rescue nil
return from_yaml.class.to_s \
  if [Symbol, String, Float, Integer].include?(from_yaml.class)
```

So now when you give the type converter `Array(String, Integer)` it converts it to `String`. This is also true for a lot of other bad inputs. Maybe it just shouldn't handle strings by default, since most things in YAML can be considered strings?

**Comment by AaronC81** [comment]
Merged in #39
